pkg/storage/fs/lister.go

```diff
@@ -3,12 +3,35 @@ package fs
 import (
 	"context"
 	"os"
+	"path/filepath"
 
+	"github.com/coreos/go-semver/semver"
 	"github.com/gomods/athens/pkg/errors"
 	"github.com/gomods/athens/pkg/observ"
 	"github.com/spf13/afero"
 )
 
+func (l *storageImpl) isSemVerDir(module string, verFileInfo os.FileInfo) (bool, error) {
+	if !verFileInfo.IsDir() {
+		return false, nil
+	}
+	ver := verFileInfo.Name()
+	if ver[0] != 'v' {
+		return false, nil
+	}
+	if _, err := semver.NewVersion(ver[1:]); err != nil {
```

Comment:
Go's interpretation of SemVer is not the same as the Spec which is what core-os I believe follows. Therefore, this could lead to some false negatives. You can check out godoc.org/golang.org/x/mod/semver for Go's interpretation.